File: core/core-api/src/main/java/com/coffee/gu/api/controller/v1/response/order/OrderListResponse.java
package com.coffee.gu.api.controller.v1.response.order;

import com.coffee.gu.order.OrderSummary;
import com.coffee.gu.enums.OrderState;

import java.math.BigDecimal;
import java.util.List;

public record OrderListResponse(
        String key,
        String name,
        BigDecimal totalPrice,
        OrderState state
) {
    public static OrderListResponse from(OrderSummary order) {
        return new OrderListResponse(
          order.key(),
          order.name(),
          order.totalPrice(),
          order.state()
        );
    }

    public static List<OrderListResponse> from(List<OrderSummary> orders) {
        return orders.stream()
                .map(OrderListResponse::from)
                .toList();
    }
}

File: core/core-api/src/main/kotlin/com/coffee/gu/api/controller/v1/response/order/OrderListResponse.kt
package com.coffee.gu.api.controller.v1.response.order

import com.coffee.gu.enums.OrderState
import com.coffee.gu.order.OrderSummary
import java.math.BigDecimal

class OrderListResponse(
    val key: String,
    val name: String,
    val totalPrice: BigDecimal,
    val state: OrderState,
) {
    companion object {
        @JvmStatic
        fun from(order: OrderSummary): OrderListResponse {
            return OrderListResponse(
                key = order.key,
                name = order.name,
                totalPrice = order.totalPrice,
                state = order.state
            )
        }

        @JvmStatic
        fun from(orders: List<OrderSummary>): List<OrderListResponse> {
            return orders.map { from(it) }
        }
    }
}
